Editor/Properties/SnappablePropertyEditor.cs

```diff
@@ -144,8 +145,31 @@ private GameObject SaveSnapZonePrefab(GameObject snapZoneBlueprint)
             snapZoneBlueprint.transform.position = Vector3.zero;
             snapZoneBlueprint.transform.rotation = Quaternion.identity;
 
-            string prefabName = snapZoneBlueprint.name.Replace("_Clone", "Highlight.prefab");
-            return PrefabUtility.SaveAsPrefabAsset(snapZoneBlueprint, $"{PrefabPath}/{prefabName}");
+            string prefabName = NamePrefab(snapZoneBlueprint.name);
+            string prefabPath = $"{PrefabPath}/{prefabName}";
+            GameObject prefab = PrefabUtility.SaveAsPrefabAsset(snapZoneBlueprint, prefabPath);
+
+            if (prefab != null)
+            {
+                Debug.LogWarningFormat("A new Snap Zone prefab was saved at {0}", prefabPath);
+            }
+            
+            return prefab;
+        }
+
+        private string NamePrefab(string originalName)
+        {
+            int invalidCharacterIndex;
+            StringBuilder prefabName = new StringBuilder(originalName);
+
+            // Unity replaces invalid characters with '_' when creating new prefabs in the editor.
+            // We try to simulate that behavior.
+            while ((invalidCharacterIndex = prefabName.ToString().IndexOfAny(Path.GetInvalidFileNameChars())) >= 0)
```

Comment:
I'd say just replace it with 
```csharp
foreach (char invalidCharacter in Path.GetInvalidFileNameChars())
{
    prefabName = prefabName.Replace(invalidCharacter, '_');
}
```

Also, your code shouldn't work because strings are immutable and you do not assign the result from the .Replace method.